src/aws-cpp-sdk-core/source/auth/SSOCredentialsProvider.cpp

```diff
@@ -28,15 +28,29 @@ using Aws::Utils::Threading::ReaderLockGuard;
 
 static const char SSO_CREDENTIALS_PROVIDER_LOG_TAG[] = "SSOCredentialsProvider";
 
-SSOCredentialsProvider::SSOCredentialsProvider() : m_profileToUse(GetConfigProfileName())
+SSOCredentialsProvider::SSOCredentialsProvider() : SSOCredentialsProvider(GetConfigProfileName(), nullptr)
 {
-    AWS_LOGSTREAM_INFO(SSO_CREDENTIALS_PROVIDER_LOG_TAG, "Setting sso credentials provider to read config from " <<  m_profileToUse);
 }
 
-SSOCredentialsProvider::SSOCredentialsProvider(const Aws::String& profile) : m_profileToUse(profile),
-                                                                             m_bearerTokenProvider(profile)
+SSOCredentialsProvider::SSOCredentialsProvider(const Aws::String& profile) : SSOCredentialsProvider(profile, nullptr)
 {
-    AWS_LOGSTREAM_INFO(SSO_CREDENTIALS_PROVIDER_LOG_TAG, "Setting sso credentials provider to read config from " <<  m_profileToUse);
+}
+
+SSOCredentialsProvider::SSOCredentialsProvider(const Aws::String& profile, std::shared_ptr<const Client::ClientConfiguration> config) :
```

Comment:
MSVC CI build isnt like `Client::ClientConfiguration`

> C:/aws-sdk-cpp/src/aws-cpp-sdk-core/source/auth/SSOCredentialsProvider.cpp(39): error C2653: 'Client': is not a class or namespace name (compiling source file C:\aws-cpp-sdk-core\ub_core.cpp)
> C:/aws-sdk-cpp/src/aws-cpp-sdk-core/source/auth/bearer-token-provider/SSOBearerTokenProvider.cpp(38): error C2653: 'Client': is not a class or namespace name (compiling source file C:\aws-cpp-sdk-core\ub_core.cpp) 

Gotta make it `Aws:Client::ClientConfiguration` in both `SSOCredentialsProvider` and `SSOBearerTokenProvider`

Reply:
Fixed.